### PR TITLE
Joanna Cortes Pomeo, Mid Mod

### PR DESCRIPTION
**Pull request by JoannaCoPo**

### CODE REVIEW

**Comment by JoannaCoPo** [comment]
FYI, just for my own peace of mind I finished the _basic_ functionality over lunch on the `conclusion` branch :)

File: app/facades/nation_facade.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class NationFacade
+  def self.nation_search_details(nation = 'Fire Nation', per_page = 97, page = 1)
```
Comment (robertguturing) [inline app/facades/nation_facade.rb:2]: I'm with you I think making page retrieval by the number of results dynamic is not what you should focus on during the mid mod. I like where you head is at but for something that is assessment style not the best idea. Just return 20 results and one page by default

File: app/controllers/search_controller.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class SearchController < ApplicationController
+  def index
+    nation  = params[:nation]
+    @nation_members = NationFacade.nation_search_details(nation, 97, 1)
```
Comment (robertguturing) [inline app/controllers/search_controller.rb:4]: Facade pattern is a good refactor here

File: app/services/nation_service.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
+class NationService
+  def self.nation_search(nation, per_page, page)
+  # def self.nation_search(nation)
+    # response = conn.get("/api/v1/characters?affiliation=#{nation}&perPage=97&page=1")
+    response = conn.get("/api/v1/characters?affiliation=#{nation}&perPage=#{per_page}&page=#{page}") #make more dynamic
```
Comment (robertguturing) [inline app/services/nation_service.rb:5]: Just hardcode the per page and page here for now

**Review by robertguturing — COMMENTED** [review]
Overall this is pretty good, worry about finishing the functionality first and getting that solid before any refactoring in the future

**Pull request closed without merging**